src/nunavut/jinja/lang/c.py

```diff
@@ -16,6 +16,153 @@
 
 from nunavut.jinja.jinja2 import TemplateRuntimeError, environmentfilter, Environment
 
+# Taken from https://en.cppreference.com/w/c/keyword
+C_RESERVED_IDENTIFIERS = frozenset([
+    'asm',
+    'auto',
+    'break',
+    'case',
+    'char',
+    'const',
+    'continue',
+    'default',
+    'defined',
+    'do',
+    'double',
+    'error',
+    'else',
+    'elif',
+    'endif',
+    'enum',
+    'extern',
+    'float',
+    'for',
+    'fortran',
+    'goto',
+    'if',
+    'ifdef',
+    'ifndef',
+    'include',
+    'inline',
+    'int',
+    'line',
+    'long',
+    'pragma',
+    'register',
+    'restrict',
+    'return',
+    'short',
+    'signed',
+    'sizeof',
+    'static',
+    'struct',
+    'switch',
+    'typedef',
+    'undef',
+    'union',
+    'unsigned',
+    'void',
+    'volatile',
+    'while',
+    '_Alignas',
+    '_Alignof',
+    '_Atomic',
+    '_Bool',
+    '_Complex',
+    '_Generic',
+    '_Imaginary',
+    '_Noreturn',
+    '_Pragma',
+    '_Static_assert',
+    '_Thread_local'
```

Comment:
The following are all reserved:

- `_[A-Z_]+[a-zA-Z0-9_]*`
- `(is|to|str|mem|wcs|atomic_|cnd_|mtx_|thrd_|tss_|memory_)[a-z][a-zA-Z_0-9]*`
- `u?int[a-zA-Z_0-9]*_t`
- `E[A-Z0-9][a-zA-Z_0-9]*`
- `FE[A-Z][a-zA-Z_0-9]*`
- `U?INT[a-zA-Z_0-9]*_(MAX|MIN|C)`
- `(PRI|SCN)[a-zX][a-zA-Z_0-9]*`
- `LC_[A-Z][a-zA-Z_0-9]*`
- `SIG_?[A-Z][a-zA-Z_0-9]*`
- `TIME_[A-Z][a-zA-Z_0-9]*`
- `ATOMIC_[A-Z][a-zA-Z_0-9]*`

https://en.cppreference.com/w/c/language/identifier
https://en.cppreference.com/w/cpp/language/identifiers

Would it not be better to make this a list of regexps?

Reply:
> Would it not be better to make this a list of regexps?

It would now! Good catch. Thanks.

Reply:
> (is|to|str|mem|wcs|atomic_|cnd_|mtx_|thrd_|tss_|memory_)[a-z][a-zA-Z_0-9]*

My reading of cppreference is that this pattern is actually `(is|to|str|mem|wcs|atomic_|cnd_|mtx_|thrd_|tss_|memory_)[a-z]+`. Am I missing something?

Reply:
It says "followed by a lowercase letter", not "by lowercase letters". Although, now that I'm thinking of it, it doesn't make much sense; perhaps they meant the latter but worded the description poorly (or I am simply misreading it). I tried looking at the standard but couldn't immediately find the relevant sections. Let's go with your interpretation.

Reply:
I'll change to `^(is|to|str|mem|wcs|atomic_|cnd_|mtx_|thrd_|tss_|memory_|memory_order_)[a-z]{1}` which is more correct and probably more efficient.